daemons/controld/controld_execd.c

```diff
@@ -827,19 +827,26 @@ build_active_RAs(lrm_state_t * lrm_state, xmlNode * rsc_list)
     return FALSE;
 }
 
-static xmlNode *
-do_lrm_query_internal(lrm_state_t *lrm_state, int update_flags)
+xmlNode *
+controld_query_executor_state(const char *node_name)
 {
     xmlNode *xml_state = NULL;
     xmlNode *xml_data = NULL;
     xmlNode *rsc_list = NULL;
     crm_node_t *peer = NULL;
+    lrm_state_t *lrm_state = lrm_state_find(node_name);
+
+    if (!lrm_state) {
+        crm_err("Could not find executor state for node %s", node_name);
```

Comment:
The only existing callers pass fsa_our_uname which is guaranteed non-NULL, but it would be a good idea for future-proofing (or maybe the function should just assume fsa_our_uname)